test/test_opam_cmd.ml

```diff
@@ -40,3 +40,55 @@ let test_tag_from_archive =
     make_test ~archive:"https://other.domain.com/some/path/file-v1.2.3.tbz"
       ~expected:(Some "v1.2.3") ()
   ]
+
+let test_classify_package =
+  let make_test ~name ~package ?(dev_repo = "dummy-dev-repo") ?archive ?(pins = []) ~expected () =
+    let test_name = Printf.sprintf "classify_package: %s" name in
+    let test_fun () =
+      let actual = Duniverse_lib.Opam_cmd.classify_package ~package ~dev_repo ~archive ~pins () in
+      Alcotest.(check (pair Testable.Opam.repo (option string))) name expected actual
+    in
+    (test_name, `Quick, test_fun)
+  in
+  [ make_test ~name:"base package"
+      ~package:{ name = "ocaml"; version = None }
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"base package versioned"
+      ~package:{ name = "ocaml"; version = Some "1" }
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"empty dev-repo" ~package:{ name = "x"; version = None } ~dev_repo:""
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"no archive" ~package:{ name = "x"; version = None }
+      ~dev_repo:"host.com/path.git" ?archive:None
+      ~expected:(`Virtual, None)
+      ();
+    make_test ~name:"github dev-repo" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo" ~archive:""
+      ~expected:(`Github ("user", "repo"), None)
+      ();
+    make_test ~name:"guess tag from archive" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo" ~archive:"file-v1.tbz"
+      ~expected:(`Github ("user", "repo"), Some "v1")
+      ();
+    make_test ~name:"unexpected github url" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://github.com/user/repo/oh/my/what/is/this" ~archive:""
+      ~expected:(`Error "weird github url", None)
+      ();
+    make_test ~name:"unknown" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://host.com/some-repo" ~archive:""
+      ~expected:(`Unknown "host.com", None)
+      ();
+    make_test ~name:"no host" ~package:{ name = "x"; version = None } ~dev_repo:"nohost"
+      ~archive:""
+      ~expected:(`Error "dev-repo without host", None)
+      ();
+    make_test ~name:"git" ~package:{ name = "x"; version = None }
+      ~dev_repo:"git+https://host.com/some-repo" ~archive:"gitpaf#pouf"
+      ~expected:(`Git "gitpaf", None)
```

Comment:
See this test, I believe it uses the archive where it should use the `dev-repo` and that's a bug @avsm

Reply:
I implemented a fix and updated the test cases, let me know if it does what it should or if I got it wrong.